src/Control/Distributed/Process/Management/Internal/Trace/Primitives.hs

```diff
@@ -168,6 +169,10 @@ withRegisteredTracer act = do
   withLocalTracer $ \t -> liftIO $ Tracer.getCurrentTraceClient t sp
   currentTracer <- receiveChan rp
   case currentTracer of
-    Nothing  -> do { (Just p') <- whereis "tracer.initial"; act p' }
+    Nothing  -> do mTP <- whereis "tracer.initial"
```

Comment:
Hmmm. Although this state invariant shouldn't ever happen, I don't think we should die here. The `Debug.hs` module depends on this code, as do a lot of other modules indirectly. If there is no initial tracer, that means that one of three things has happened...

1. the initial tracer started by `startServiceProcesses` was killed
2. an exception killed `startDefaultTracer` before we could register the name
3. someone unregistered the name

Any of those cases represents a systemic problem, but fundamentally it's not an issue we should solve here. We can open a separate issue to discuss whether the node controller should monitor and restart service processes.